rocAL/rocAL/source/coco_meta_data_reader_key_points.cpp

```diff
@@ -0,0 +1,388 @@
+        joints_data_batch.score_batch.push_back(it->second->get_joints_data().score);
+        joints_data_batch.rotation_batch.push_back(it->second->get_joints_data().rotation);
+    }
+    _output->get_joints_data_batch() = joints_data_batch;
+}
+
+void COCOMetaDataReaderKeyPoints::add(std::string image_id, ImgSizes image_size, JointsData joints_data)
+{
+    pMetaDataKeyPoint info = std::make_shared<KeyPoint>(image_size, joints_data);
+    _map_content.insert(pair<std::string, std::shared_ptr<KeyPoint>>(image_id, info));
+}
+
+void COCOMetaDataReaderKeyPoints::print_map_contents()
+{
+    JointsData joints_data;
+    for (auto &elem : _map_content)
+    {
+        std::cout << "\nName :\t " << elem.first<<std::endl;
+        joints_data = elem.second->get_joints_data();
+        std::cout << "ImageID: " << joints_data.image_id << std::endl;
+        std::cout << "AnnotationID: " << joints_data.annotation_id << std::endl;
+        std::cout << "ImagePath: "<< joints_data.image_path<<std::endl;   
+        std::cout << "center (x,y) : " << joints_data.center[0] << " " << joints_data.center[1] << std::endl;
+        std::cout << "scale (w,h) : " << joints_data.scale[0] << " " << joints_data.scale[1] << std::endl;
+        for (unsigned int i = 0; i < NUMBER_OF_JOINTS; i++)
+        {
+            std::cout << " x : " << joints_data.joints[i][0] << " , y : " << joints_data.joints[i][1] << " , v : " << joints_data.joints_visibility[i][0] << std::endl;
+        }
+        std::cout << "Score: " <<  joints_data.score << std::endl;
+        std::cout << "Rotation: " <<  joints_data.rotation << std::endl;
+    }
+}
+
+void COCOMetaDataReaderKeyPoints::read_all(const std::string &path)
+{
+    _coco_metadata_read_time.start(); // Debug timing
+    std::ifstream f;
+    f.open(path, std::ifstream::in | std::ios::binary);
+    if (f.fail())
+        THROW("ERROR: Given annotations file not present " + path);
+    f.ignore(std::numeric_limits<std::streamsize>::max());
+    auto file_size = f.gcount();
+    f.clear(); //  Since ignore will have set eof.
+    if (file_size == 0)
+    { // If file is empty return
+        f.close();
+        THROW("ERROR: Given annotations file not valid " + path);
+    }
+    std::unique_ptr<char, std::function<void(char *)>> buff(
+        new char[file_size + 1],
+        [](char *data)
+        { delete[] data; });
+    f.seekg(0, std::ios::beg);
+    buff.get()[file_size] = '\0';
+    f.read(buff.get(), file_size);
+    f.close();
+
+    LookaheadParser parser(buff.get());
+
+    ImgSizes img_sizes;
+    JointsData joints_data;
+
+    ImgSize img_size;
+    std::vector<float> box_center, box_scale;
+    float score = 1.0;
+    float rotation = 0.0;
+    float aspect_ratio = (_out_img_width * 1.0 / _out_img_height);
+    float inverse_aspect_ratio = 1 / aspect_ratio;
+    float inverse_pixel_std = 1 / (1.0 * PIXEL_STD); 
```

Comment:
please remove multiplication by 1.0. Can pixel_std be variable in any cases?

Reply:
Done
pixel_std will be fixed for all the cases